src/Surfnet/Stepup/Tests/Configuration/InstitutionConfigurationTest.php

```diff
@@ -0,0 +1,264 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\Stepup\Tests\Configuration;
+
+use PHPUnit_Framework_TestCase as TestCase;
+use Rhumsaa\Uuid\Uuid;
+use Surfnet\Stepup\Configuration\Entity\RaLocation;
+use Surfnet\Stepup\Configuration\InstitutionConfiguration;
+use Surfnet\Stepup\Configuration\Value\ContactInformation;
+use Surfnet\Stepup\Configuration\Value\Institution;
+use Surfnet\Stepup\Configuration\Value\InstitutionConfigurationId;
+use Surfnet\Stepup\Configuration\Value\Location;
+use Surfnet\Stepup\Configuration\Value\RaLocationId;
+use Surfnet\Stepup\Configuration\Value\RaLocationList;
+use Surfnet\Stepup\Configuration\Value\RaLocationName;
+
+class InstitutionConfigurationTest extends TestCase
+{
+    /**
+     * @test
+     * @group aggregate
+     */
+    public function an_institution_configuration_can_be_created_with_a_given_institution_configuration_id_and_a_given_institution()
+    {
+        $institution = new Institution('Test institution');
+        $institutionConfigurationId = InstitutionConfigurationId::from($institution);
+
+        $institutionConfiguration = InstitutionConfiguration::create($institutionConfigurationId, $institution);
+        $expectedRaLocations = new RaLocationList([]);
+
+        $this->assertEquals($institutionConfigurationId, $institutionConfiguration->getInstitutionConfigurationId());
+        $this->assertEquals($institution, $institutionConfiguration->getInstitution());
+        $this->assertEquals($expectedRaLocations, $institutionConfiguration->getRaLocations());
+    }
+
+    /**
+     * @test
+     * @group aggregate
+     */
+    public function an_ra_location_can_be_added_to_an_institution_configuration()
+    {
+        $institution = new Institution('Test institution');
+        $institutionConfigurationId = InstitutionConfigurationId::from($institution);
+
+        $raLocationId       = new RaLocationId(self::uuid());
+        $raLocationName     = new RaLocationName('An RA location name');
+        $location           = new Location('A location');
+        $contactInformation = new ContactInformation('Contact information');
+
+        $expectedRaLocations = new RaLocationList(
+            [
+                RaLocation::create(
+                    $raLocationId,
+                    $raLocationName,
+                    $location,
+                    $contactInformation
+                ),
+            ]
+        );
+
+        $institutionConfiguration = InstitutionConfiguration::create($institutionConfigurationId, $institution);
+        $institutionConfiguration->addRaLocation(
+            $raLocationId,
+            $raLocationName,
+            $location,
+            $contactInformation
+        );
+
+        $actualRaLocations = $institutionConfiguration->getRaLocations();
```

Comment:
I agree that events are the way this object communicates and I started out that way. However, it is a messy approach as you would have to get the uncommitted events from the aggregate. For each `DomainMessage` in there, the payload should be as expected. I do see value in asserting the right (sequence of) event(s) is produced.

I also see value in testing the state change within the object. A certain state change is expected when a certain method is called. These kinds of expectations should be tested. Whether or not this is done via CQRS or otherwise, to me, seems of less importance as this kind of behaviour is part of the unit under test.